File: articles/grit-and-conscientiousness-overlap-distinction-and-debate/go/grit_conscientiousness_score.go
package main

import "fmt"

func ConscientiousnessScore(industriousness, orderliness, dependability, responsibility, achievementStriving float64) float64 {
	return 0.30*industriousness +
		0.18*orderliness +
		0.18*dependability +
		0.17*responsibility +
		0.17*achievementStriving
}

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func main() {
	c := ConscientiousnessScore(0.81, 0.52, 0.67, 0.74, 0.79)
	g := GritScore(0.91, 0.63)

	fmt.Printf("Synthetic conscientiousness score: %.3f\n", c)
	fmt.Printf("Synthetic grit score: %.3f\n", g)
}

File: articles/grit-and-conscientiousness-overlap-distinction-and-debate/fortran/grit_conscientiousness_score.f90
program grit_conscientiousness_score
  implicit none
  real :: industriousness, orderliness, dependability, responsibility, achievement_striving
  real :: perseverance_effort, consistency_interests
  real :: conscientiousness, grit

  industriousness = 0.81
  orderliness = 0.52
  dependability = 0.67
  responsibility = 0.74
  achievement_striving = 0.79
  perseverance_effort = 0.91
  consistency_interests = 0.63

  conscientiousness = 0.30 * industriousness + 0.18 * orderliness + 0.18 * dependability + &
                      0.17 * responsibility + 0.17 * achievement_striving
  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests

  print *, "Synthetic conscientiousness score:", conscientiousness
  print *, "Synthetic grit score:", grit
end program grit_conscientiousness_score
